public/posts/v1.4.md

```diff
@@ -0,0 +1,268 @@
+You additionally have to add the `--unstable` flag because this feature is not
+stable yet.
+
+```shell
+$ echo "console.log('Hello World!')" > mod.ts
+$ deno run --watch mod.ts
+Check file:///home/deno/mod.ts
+Hello World
+Watcher Process terminated! Restarting on file change...
+# now run `echo "console.log('File watching works!')" > ./mod.ts` in a different terminal
+Watcher File change detected! Restarting!
+Check file:///home/deno/mod.ts
+File watching works!
+Watcher Process terminated! Restarting on file change...
+```
+
+The watch flag takes no arguments for directories or files to watch. Instead it
+automatically determines all of the local imports of your script, and watches
+those.
+
+Currently file watching is only supported for `deno run`, but in the future it
+will also be added to `deno test` and possibly other subcommands.
+
+## `deno test --coverage`
+
+You can now find code that is not covered by your tests using the `--coverage`
+flag for `deno test`. When enabled this will print a summary of your code
+coverage per file after all tests are run. You additionally have to add the
+`--unstable` flag because this feature is not stable yet.
+
+```
+$ git clone git@github.com:denosaurs/deno_brotli.git && cd deno_brotli
+$ deno test --coverage --unstable
+Debugger listening on ws://127.0.0.1:9229/ws/5a593019-d185-478b-a928-ebc33e5834be
+Check file:///home/deno/deno_brotli/.deno.test.ts
+running 2 tests
+test compress ... ok (26ms)
+test decompress ... ok (13ms)
+
+test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (40ms)
+
+test coverage:
+file:///home/deno/deno_brotli/mod.ts 100.000%
+file:///home/deno/deno_brotli/wasm.js 100.000%
+```
+
+Currently the only available output format is the text summary. Other output
+formats like `lcov` and `json` will be added in the future.
+
+## Stricter type checks in `--unstable`
+
+For all users using `--unstable` the `isolatedModules` and
+`importsNotUsedAsValues` TypeScript compiler options will be switched on by
+default now. We will enable these flags by default for everyone in the future.
+These flags enable some stricter checks in the TypeScript compiler that will
+likely lead to some new errors you have not seen before:
+
+```
+ERROR TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
+
+ERROR TS1371: This import is never used as a value and must use 'import type' because the 'importsNotUsedAsValues' is set to 'error'.
+```
+
+These errors occur when interfaces or type aliases are imported or
+re-exported. To fix the error, change your imports and re-exports to use
+`import type` and `export type`. Example:
+
+```ts
+// Bad
+import { MyType } from "./mod.ts";
+export { MyType } from "./mod.ts";
+
+// Good
+import type { MyType } from "./mod.ts";
+export type { MyType } from "./mod.ts";
+```
+
+These flags also disallow the use of `const enum`.
```

Comment:
This isn't true, is it?

Reply:
I am pretty sure they are meant to. If I understood @kitsonk correctly, `const enum` is a type directed emit. That actually may be another flag though...

Reply:
They're not meant to. It used to be the behaviour that _ambient_ declarations threw under `isolatedModules` but even that is changed as of https://github.com/microsoft/TypeScript/pull/28465. Non-ambient `const enum`s never threw, they were always emitted like `enum` in that case. Note the following quote in that PR:

>  With non-ambient const enums, we know that an isolatedModules build will emit the
enum and produce a plain reference rather than inlining the constant, so
everything will still work.

This is the thing I complained was not documented anywhere, but there's no reason to believe it's a bug that is actually supposed to throw. I've seen it acknowledged many times in TS issues/PRs.

Reply:
:+1: I'll remove the line.